apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/controller/ConfigsExportController.java

```diff
@@ -94,30 +58,35 @@ public void exportItems(@PathVariable String appId, @PathVariable String env,
 
     String fileName = fileNameSplit.size() <= 1 ? Joiner.on(".")
         .join(namespaceName, ConfigFileFormat.Properties.getValue()) : namespaceName;
-    NamespaceBO namespaceBO = namespaceService.loadNamespaceBO(appId, Env.fromString
+    NamespaceBO namespaceBO = namespaceService.loadNamespaceBO(appId, Env.valueOf
         (env), clusterName, namespaceName);
 
     //generate a file.
-    res.setHeader("Content-Disposition", "attachment;filename=" + fileName);
-
-    List<String> fileItems = namespaceBO.getItems().stream().map(itemBO -> {
-      String key = itemBO.getItem().getKey();
-      String value = itemBO.getItem().getValue();
-      if (ConfigConsts.CONFIG_FILE_CONTENT_KEY.equals(key)) {
-        return value;
-      }
-
-      if ("".equals(key)) {
-        return Joiner.on("").join(itemBO.getItem().getKey(), itemBO.getItem().getValue());
-      }
-
-      return Joiner.on(" = ").join(itemBO.getItem().getKey(), itemBO.getItem().getValue());
-    }).collect(Collectors.toList());
-
+    res.setHeader(HttpHeaders.CONTENT_DISPOSITION, "attachment;filename=" + fileName);
+    // file content
+    final String configFileContent = NamespaceBOUtils.convert2configFileContent(namespaceBO);
     try {
-      ConfigToFileUtils.itemsToFile(res.getOutputStream(), fileItems);
+      // write content to net
+      res.getOutputStream().write(configFileContent.getBytes());
     } catch (Exception e) {
       throw new ServiceException("export items failed:{}", e);
     }
   }
+
+  /**
+   * Export all configs in a compressed file.
+   * Just export namespace which current exists read permission.
+   * The permission check in service.
+   */
+  @GetMapping("/export")
+  public void exportBy(
```

Comment:
rename it to exportAll?

Reply:
why is the export function targed to all environments but the import function only targeted to some specific env?

Reply:
> why is the export function targed to all environments but the import function only targeted to some specific env?

The appId, cluster, namespace are parsed from config filename.

A yml config filename like follow:
![image](https://user-images.githubusercontent.com/15523186/83347379-40a10c80-a357-11ea-8549-ae71fd9913d5.png)
appId, cluster, namespace are divided by character `+`.

---

Need to a **specific env** because the filename doesn't contain the information of *env*.

The information of *env* can be add to filename.

After add *env*，the filename is changed like follow:

![image](https://user-images.githubusercontent.com/15523186/83347506-fe2bff80-a357-11ea-8823-44ad02c8c9be.png)

The main idea it that parse the information from filename instead of user's choose them by UI operation.

---

Should users add *env* to config filename, or they choose *env* by UI?

Here I follow the config's local cache filename which apollo client stored , so the information of *env* is not added.

But we can let users add *env* to their config filename, so users don't need to choose *env* by UI.

Which one is better?

Reply:
I think it may be better to input more information on the UI